File: src/main/java/br/anhembi/funmodechild/entity/Category.java
package br.anhembi.funmodechild.entity;

import br.anhembi.funmodechild.model.response.CategoryResponse;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document("categories")
public class Category {

    @Id
    private String id;
    private String name;

    public CategoryResponse toApiResponse() {
        return new CategoryResponse(this.id, this.name);
    }
}

File: src/main/kotlin/br/anhembi/funmodechild/entity/Category.kt
package br.anhembi.funmodechild.entity

import br.anhembi.funmodechild.model.response.CategoryResponse
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document("categories")
data class Category(
    @Id
    val id: String? = null,
    val name: String
) {
    fun toApiResponse() = CategoryResponse(
        id = this.id!!,
        name = this.name
    )
}
